tests/logging_tests/tests.py

```diff
@@ -109,16 +109,82 @@ def test_django_logger_debug(self):
         self.assertEqual(self.logger_output.getvalue(), '')
 
 
+class LoggingAssertionMixin(object):
+    def assert_request_logs(self, url, level, msg, status_code,
+                            logger='django.request'):
+        with patch_logger(logger, level, log_kwargs=['extra']) as calls:
+            self.client.get(url)
+        self.assertEqual(len(calls), 1,
+                         "Wrong number of call for logger '{}' in '{}' level".format(logger, level))
+        self.assertEqual(calls[0][0], msg)
+        self.assertEqual(calls[0][1]['extra']['status_code'], status_code)
+
+
 @override_settings(DEBUG=True, ROOT_URLCONF='logging_tests.urls')
-class HandlerLoggingTests(SetupDefaultLoggingMixin, LoggingCaptureMixin, SimpleTestCase):
+class HandlerLoggingTests(SetupDefaultLoggingMixin, LoggingAssertionMixin, LoggingCaptureMixin, SimpleTestCase):
 
     def test_page_found_no_warning(self):
         self.client.get('/innocent/')
         self.assertEqual(self.logger_output.getvalue(), '')
 
     def test_page_not_found_warning(self):
-        self.client.get('/does_not_exist/')
-        self.assertEqual(self.logger_output.getvalue(), 'Not Found: /does_not_exist/\n')
+        self.assert_request_logs(
+            url='/does_not_exist/',
+            level='warning',
+            status_code=404,
+            msg='Not Found: /does_not_exist/',
+        )
+
+    def test_page_not_found_raised(self):
+        self.assert_request_logs(
+            url='/does_not_exist_raised/',
+            level='warning',
+            status_code=404,
+            msg='Not Found: /does_not_exist_raised/',
+        )
+
+    def test_uncaught_exception(self):
+        try:
+            self.assert_request_logs(
+                url='/uncaught_exception/',
+                level='error',
+                status_code=500,
+                msg='Internal Server Error: /uncaught_exceptio/',
+            )
+        except:
```

Comment:
This bare `except` will catch anything, including the exception that marks a failing test, I think :-) Better to use `assertRaises` and explicitly assert that the specific expected uncaught exception bubbles up.